### ISSUE TITLE
unreachable_code_linter hits #nolint messages

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1347]
```
function() {
  do_something
  # nolint start: one_linter.
  edge_case_stuff()
  # nolint stop: one_linter.
}
```

will throw the lint on `# nolint stop`. this seems less than ideal but I'm not sure if it can be easily fixed.

**Comment by AshesITR** [comment]
Maybe never lint `COMMENT` nodes?

**Comment by MichaelChirico** [comment]
Actually comments were one of the things I definitely wanted to lint. Often see things like

```
function() {
  do_something()
  # print(x)
}
```

i.e., vestigial comments that should probably be removed (or at least moved away from the very end of the function)

**Comment by AshesITR** [comment]
Shouldn't that be a separate lint message at least? Trailing comments aren't exactly unreachable code.

Also, just to clarify, these lints are excluded by `exclude()`, right?

**Comment by MichaelChirico** [comment]
Oh, good point. Edited the issue -- they're not excluded because they're specific to a different linter.

**Comment by AshesITR** [comment]
I'm having trouble reproducing this:

```r
expect_lint(
  trim_some("
    foo <- function() {
      do_something
      # nolint start: one_linter.
      a = 42
      # nolint end
    }
  "),
  NULL,
  list(unreachable_code_linter(), one_linter = assignment_linter())
)
```

doesn't fail on main.
